### PR TITLE
Port assignment 1 from ocw to Linux kernel labs

### PR DESCRIPTION
**Pull request by danield20**
The statement of assignment 1 is ported from ocw to Linux kernel labs and the content is translated to English.

### CODE REVIEW

File: Documentation/teaching/so2/assign1-kprobe-based-tracer.rst [hunk 1]
```diff
@@ -0,0 +1,173 @@
+==================================
+Assignment 1 - Kprobe based tracer
+==================================
+
+-  Deadline: :command:`Wednesday, 7 April 2021, 23:00`
```
Comment (claudiughioc) [inline Documentation/teaching/so2/assign1-kprobe-based-tracer.rst:5]: April 7th

File: Documentation/teaching/so2/assign1-kprobe-based-tracer.rst [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+For the ``kmalloc`` and ``kfree`` calls the total quantity of allocated and deallocated memory will be
+shown.
+
+The surveillant will be implemented as a kernel module with the name ``tracer.ko``.
+
+Implementation details
+----------------------
+
+The interception will be done by recording a sample (``kretprobe``) for each of the above functions. The
+surveillant will retain a list/table of dispersion with the monitored processes and will account for
```
Comment (claudiughioc) [inline Documentation/teaching/so2/assign1-kprobe-based-tracer.rst:36]: will retain a list/hashtable

File: Documentation/teaching/so2/assign1-kprobe-based-tracer.rst [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+the above information for these processes.
+
+For the control of the list/table of dispersion with the monitored processes, a char device called ``/dev/tracer``
+will be used, with major 10 and minor 42. It will expose an ``ioctl`` interface with two arguments:
```
Comment (claudiughioc) [inline Documentation/teaching/so2/assign1-kprobe-based-tracer.rst:40]: hightlight 10 and 42

File: Documentation/teaching/so2/assign1-kprobe-based-tracer.rst (hunk 2 above)
Comment (claudiughioc) [reply Documentation/teaching/so2/assign1-kprobe-based-tracer.rst:36]: replace with hashtable in all occurences

**Review by claudiughioc — APPROVED** [review]

File: Documentation/teaching/so2/assign1-kprobe-based-tracer.rst (hunk 1 above)
Comment (danield20) [reply Documentation/teaching/so2/assign1-kprobe-based-tracer.rst]: Done.

File: Documentation/teaching/so2/assign1-kprobe-based-tracer.rst (hunk 2 above)
Comment (danield20) [reply Documentation/teaching/so2/assign1-kprobe-based-tracer.rst]: Done.

File: Documentation/teaching/so2/assign1-kprobe-based-tracer.rst (hunk 3 above)
Comment (danield20) [reply Documentation/teaching/so2/assign1-kprobe-based-tracer.rst]: Done.

File: Documentation/teaching/so2/assign1-kprobe-based-tracer.rst [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+-  Deadline: :command:`Wednesday, April 7th 2021, 23:00`
+
+Assignment's Objectives
+=======================
+
+*  acquiring knowledge related to the instrumentation of functions in the Linux kernel (``kretprobes`` mechanism)
+*  gaining knowledge regarding the ``/proc`` file system from the Linux kernel
+*  getting more familiar with the data structures specific to the Linux kernel (``hash table``, ``list``)
+
```
Comment (dbaluta) [inline Documentation/teaching/so2/assign1-kprobe-based-tracer.rst:13]: I would replace `acuiqiring` with `aquire`, `gaining` with `gain`, etc. 

Also, last bullet I would rephrase it like this:
* get familiar with data structure specific to Linux kernel (``hash_table`` and ``list``).

Comment (danield20) [reply Documentation/teaching/so2/assign1-kprobe-based-tracer.rst:13]: Done.

**Review by dbaluta — APPROVED** [review]
